types/lodash/index.d.ts

```diff
@@ -9515,7 +9515,7 @@ declare namespace _ {
          * @param value The value to check.
          * @return Returns true if value is correctly classified, else false.
          */
-        isFunction(value?: any): value is ((...args: any[]) => any) | Function;
+        isFunction(value?: any): value is (...args: any[]) => any;
```

Comment:
I think we should make the parameter non-optional (remove the `?`).

Also don't forget to fix `lodash-tests.ts`. Probably change `let result: number = value;` to `let result: number | Function = value;`, and add the test case I mentioned in my other comment.